### PR TITLE
fix(ci): re-enable kill-meta in deterministic simulation test

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Followup 1-3 of https://github.com/risingwavelabs/risingwave/pull/18678#issuecomment-2395692599

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Fails background_ddl recovery test. We can fix that separately.

**Review by stdrc — APPROVED** [review]
Slightly prefer separating the changes for upgrading deps and re-enabling, but not going to block this. Approve on behalf of Cargo-lock.

**Comment by kwannoel** [comment]
Blocked, need some help for: https://github.com/madsim-rs/madsim/issues/228. Opened an issue in the upstream madsim repo.

File: ci/scripts/deterministic-recovery-test.sh [hunk 1]
```diff
@@ -44,7 +44,7 @@ filter_stack_trace_for_all_logs() {
   done
 }
 
-trap filter_stack_trace_for_all_logs ERR
+# trap filter_stack_trace_for_all_logs ERR
```
Comment (fuyufjh) [inline ci/scripts/deterministic-recovery-test.sh:47]: Why?

Comment (kwannoel) [reply ci/scripts/deterministic-recovery-test.sh:47]: `filter_stack_trace_for_all_logs` accidentally removes some useful stacktrace from madsim.

But without it, the stacktrace dump in the log file is really large, and makes the file size few hundred MB to 1GB+ previously.

So I temporarily disable it here. This change won't be merged in.

**Pull request closed without merging**